desc/compute/geom_utils.py

```diff
@@ -45,7 +46,7 @@ def rotation_matrix(axis, angle=None):
     R1 = jnp.cos(angle) * jnp.eye(3)
     R2 = jnp.sin(angle) * jnp.cross(axis, jnp.identity(axis.shape[0]) * -1)
     R3 = (1 - jnp.cos(angle)) * jnp.outer(axis, axis)
-    return R1 + R2 + R3
+    return R1 + jnp.nan_to_num(R2 + R3)  # if axis=[0, 0, 0], R2 & R3 are NaN
```

Comment:
`FourierPlanarCurve` is a 2D curve in the X-Y plane, and get rotated into another plane in 3D space. If the curve stays in the X-Y plane (as is the default parameters), then this function gets called with `axis=[0, 0, 0]` and `angle=0`.

Reply:
Can you check that reverse mode ad works for this? It sometimes has weirdness with overwriting nans

Reply:
This actually doesn't work with forward or reverse mode AD. I spent a bunch of time thinking about this and couldn't come up with a good solution, but I welcome help if you have an idea. 

I still think this PR is worth merging even if we don't fix that issue because: 

1. The old code was returning incorrect values
2. The old code also did not properly handle the case `axis=[0, 0, 0]`
3. In practice this will only arise for vertical field coils, and I don't think that is a high-priority use case